charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -195,7 +195,19 @@ receivers:
 {{- end }}
 
 {{- define "opentelemetry-collector.applyClusterMetricsConfig" -}}
-{{- $config := mustMergeOverwrite (dict "service" (dict "pipelines" (dict "metrics" (dict "receivers" list)))) (include "opentelemetry-collector.clusterMetricsConfig" .Values | fromYaml) .config }}
+{{- $config := .config }}
+
+{{- /* Add leader election extension if enabled */ -}}
+{{- if .Values.presets.clusterMetrics.enableLeaderElection }}
+{{- $config = mustMergeOverwrite (dict "extensions" (dict "k8s_leader_elector" (dict "lease_name" "otel-collector-leader" "lease_namespace" (include "opentelemetry-collector.namespace" .)))) $config }}
```

Comment:
Please break out the k8s leader election config into its own helper template like `opentelemetry-collector.clusterMetricsConfig` does.